### PR TITLE
[BugFix] Fix state-dict

### PR DESCRIPTION
**Pull request by vmoens**
## Description

Fixes calls to state-dict to make them compatible with nn.Module.

Introduces related tests and docstring.

cc @matteobettini

### CODE REVIEW

**Review by matteobettini — APPROVED** [review]
LGTM

maybe one test where the loaded data is not all zeros just to make sure

**Comment by matteobettini** [comment]
we also may one some tests in torchrl to test the non regression of saving the state dict of loss modules

**Comment by vmoens** [comment]
> LGTM
> 
> maybe one test where the loaded data is not all zeros just to make sure

We load onto zeroed data no?

**Comment by vmoens** [comment]
> we also may want some tests in torchrl to test the non regression of saving the state dict of loss modules

on it already

**Comment by vmoens** [comment]
> I meant something like getting the params from a linear module, saving, and reloading and checking against the original params

Ah but we do that, what is the problem with doing it with zero?
